src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -571,10 +593,20 @@ internal static bool IsBetterDeletionMatch(FilterResult result1, FilterResult re
             => result1.CompareTo(result2) > 0;
 
         internal static bool MatchesFilterText(
-            CompletionHelper helper, RoslynCompletionItem item,
-            string filterText, CompletionTriggerKind initialTriggerKind,
-            CompletionFilterReason filterReason, ImmutableArray<string> recentItems)
+            CompletionHelper helper,
+            RoslynCompletionItem item,
+            string filterText,
+            CompletionTriggerKind initialTriggerKind,
+            CompletionFilterReason filterReason,
+            ImmutableArray<string> recentItems,
+            out PatternMatch? patternMatch)
         {
+            // Get the match of the given completion item for the pattern provided so far. 
+            // A completion item is checked against the pattern by see if it's 
+            // CompletionItem.FilterText matches the item. That way, the pattern it checked 
+            // against terms like "IList" and not IList<>
+            patternMatch = helper.GetMatch(item.FilterText, filterText, includeMatchSpans: true, CultureInfo.CurrentCulture);
```

Comment:
>patternMatch = helper.GetMatch(item.FilterText, filterText, includeMatchSpans: true, CultureInfo.CurrentCulture); [](start = 12, length = 113)

perf concern. 

I like the idea from functional point of view. It provides an access to the patternMatch in all later comparisons. 

However, there are situations when we didn't make any calculations of pattern matches at all, e.g. for deletions. Now we do. Some scenarios may run slowly.

Reply:
This is the only major concern I have. I attempted to this particular changes a couple times but it stopped me.

If you can assure it is safe from a perf perspective, I totally support the change. I like the idea.

Also welcome comments from @CyrusNajmabadi 

---
In reply to: [321439499](https://github.com/dotnet/roslyn/pull/38499#discussion_r321439499) [](ancestors = 321439499)

Reply:
Good point. I haven't done any perf analysis yet, will add result once it's done. But my intuition is that in regular typing scenario, the extra sorting cost could be somewhat mitigated by decreased number of calling GetMatch. Will definitely keep an eye on deletion scenario too.